testing/e2e-test-utilities/src/main/kotlin/net/corda/e2etest/utilities/AssertWithRetryBuilder.kt

```diff
@@ -46,6 +46,21 @@ class AssertWithRetryBuilder(private val args: AssertWithRetryArgs) {
     }
 }
 
+private data class Attempt(val attemptNumber: Int, val timeTried: Duration, val response: String)
+
+private fun formatAttempts(attempts: Iterable<Attempt>): String =
```

Comment:
Minor: I would change to an extension function, like so:
```suggestion
private fun Iterable<Attempt>.prettyPrint(): String =
```